### PR TITLE
Paraphraser kpi

### PR DESCRIPTION
**Pull request by puleon**
- paraphraser model with elmo tuned embeddings
- paraphraser model pretrained on the collected news dataset
- paraphraser model tuned on the paraphraser dataset
- scores for the paraphraser models
- tests for all configs, except for deeppavlov/configs/elmo/elmo_paraphraser_fine_tuning.json

### CODE REVIEW

File: deeppavlov/configs/elmo/elmo_paraphraser_fine_tuning.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    "requirements": [
+      "{DEEPPAVLOV_PATH}/requirements/tf.txt",
+      "{DEEPPAVLOV_PATH}/requirements/tf-hub.txt"
+    ],
+    "download": [
+      {
+        "url": "http://files.deeppavlov.ai/deeppavlov_data/paraphraser_train_and_pretrain_texts.tar.gz",
+        "subdir": "{DOWNLOADS_PATH}/paraphraser_train_and_pretrain_texts"
+      }
```
Comment (yoptar) [inline deeppavlov/configs/elmo/elmo_paraphraser_fine_tuning.json]: If this config is used for fine-tuning an elmo model, shouldn't it download the pretrained one first?

File: deeppavlov/configs/ranking/paraphrase_ident_paraphraser.json [hunk 2]
```diff
@@ -79,9 +79,9 @@
     "pytest_max_batches": 2,
     "train_metrics": ["f1", "acc", "log_loss"],
     "metrics": ["f1", "acc", "log_loss"],
-    "validation_patience": 10,
-    "val_every_n_epochs": 5,
-    "log_every_n_batches": 12
+    "validation_patience": 50,
```
Comment (yoptar) [inline deeppavlov/configs/ranking/paraphrase_ident_paraphraser.json]: Are you sure about validation_patience of 50? That's a lot

File: deeppavlov/configs/elmo/elmo_paraphraser_fine_tuning.json (hunk 1 above)
Comment (puleon) [reply deeppavlov/configs/elmo/elmo_paraphraser_fine_tuning.json]: added download.

File: deeppavlov/configs/ranking/paraphrase_ident_paraphraser.json (hunk 2 above)
Comment (puleon) [reply deeppavlov/configs/ranking/paraphrase_ident_paraphraser.json]: fixed.

File: deeppavlov/dataset_readers/siamese_reader.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.commands.utils import expand_path
+
+
+@register('siamese_reader')
+class SiameseReader(DatasetReader):
+    """The class to read dataset for ranking or paraphrase identification with Siamese networks.
+
+    Args:
+        data_path: A path to a folder with dataset files.
```
Comment (nikolay-bushkov) [inline deeppavlov/dataset_readers/siamese_reader.py]: The class docstring could contain args for __ init__() method

**Pull request merged**